core/persist/PersistenceProvider.ts

```diff
@@ -62,57 +64,67 @@ export abstract class PersistenceProvider<S> {
      * Throws if unable to perform initial read operation.
      *
      * Note:
-     * - `destroy()` must be called when the provider is no longer needed.
+     * - `destroy()` must be called when the provider is no longer needed unless an owner is provided.
```

Comment:
why aren't target and owner just both required for simplicity (where owner can be defaulted to Target, if its a HoistBase).  Check and throw at runtime.